RELEASE_NOTES.md

```diff
@@ -0,0 +1,57 @@
+# 3.4 
+
+The top four improvements in Dafny 3.4 are:
+- For certain classes of changes, prevent unexpected changes in verification behavior when changing a Dafny program 
+- Add command line options to assist in debugging verification performance.
+- Critical fixes to the IDE and greatly improved responsiveness of non-verification IDE features. 
+- The C# back-end supports traits as type parameters on datatypes.
```

Comment:
Add a note that compiled downcasts are supported for these only for co-variant and non-variant type parameters, not yet contra-variant type parameters. This same note also applies under the `C#` section below (or perhaps this detail should only be mentioned in the `C#` section below, keeping this list of "top improvements" simpler to read).